### ISSUE TITLE
print detailed exception info to server console on 500 errors

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#1861]
Currently, the web server middleware swallows all Python exceptions and returns the traceback information to the client in a 500 error. This makes debugging difficult.

It should be printed to the console and saved to log files.

**Comment by rtibbles** [comment]
Is this for devserver or production server?

**Comment by indirectlylit** [comment]
probably both? might be helpful for folks reporting bugs from the "real world"

**Comment by rtibbles** [comment]
500s should produce a visible traceback in dev mode already, which is why I was checking if this issue was about that not working, or enhancing the reporting in production.

**Comment by indirectlylit** [comment]
Seems to not be producing a traceback in devmode. Steps to repro:

Update the Learn view.py to raise an exception:

```python
class LearnView(TemplateView):
    template_name = "learn/learn.html"

    def get_context_data(self, **kwargs):
        raise Exception("hell")
```

output:

```
DEBUG    Debug mode is on!
Performing system checks...

System check identified no issues (1 silenced).
September 03, 2017 - 15:53:29
Django version 1.9.7, using settings 'kolibri.deployment.default.settings.base'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
[03/Sep/2017 15:53:30] "GET /learn/ HTTP/1.1" 500 80896
```

**Comment by rtibbles** [comment]
And no traceback in the browser?

**Comment by rtibbles** [comment]
Looks like we should probably add this in dev mode: https://docs.djangoproject.com/en/1.11/topics/email/#console-backend

In Chrome I have always been able to see these error pages on the preview tab of the network response in question, however it is a bit small by default.

**Comment by indirectlylit** [comment]
Sorry - to clarify, the browser-based error pages work as expected. Those I would expect to be turned off in a production environment.

This issue was only about adding stack traces to the server console and logs.

> In Chrome I have always been able to see these error pages on the preview tab of the network response in question

good point, I always forget that

**Comment by rtibbles** [comment]
Great, think I've got it. The stack traces are written to logs, so that's covered already, so just adding 500 stacktraces to terminal output in dev mode I think is the main issue here.

**Comment by benjaoming** [comment]
All 500 errors from requests are logged already in `~/.kolibri/kolibri.log`

Notice that for technical reasons, we have `.kolibri/server.log` for uncaught daemon process outputs and `.kolibri/kolibri.log` for log rotation and output from Python's logging system.

**Comment by indirectlylit** [comment]
whoops - yes you're both correct that the 500s are logged to disk. I had incorrectly assumed that the terminal output and logs were equivalent

**Comment by rtibbles** [comment]
Fixed in #2113
